test/uk/gov/hmrc/eusubsidycompliance/models/SubsidyUpdateSpec.scala

```diff
@@ -0,0 +1,50 @@
+package uk.gov.hmrc.eusubsidycompliance.models
+
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import play.api.libs.json.Json
+import uk.gov.hmrc.eusubsidycompliance.models.types.{EORI, EisSubsidyAmendmentType, SubsidyAmount, SubsidyRef, TraderRef, UndertakingRef}
+
+import java.time.LocalDate
+
+class SubsidyUpdateSpec extends AnyWordSpec with Matchers {
```

Comment:
Handy having a spec to see the json contracts we allow. Relying on case classes to generate test data means are contracts are very hard to work out, especially nested case classes.